graphs/uniform_search_cost.py

```diff
@@ -0,0 +1,260 @@
+                        min_cell = dist[row + dr][col + dc]
+                        next_cell = [row + dr, col + dc]
+            curr_node = next_cell
+            if curr_node is None:
+                break
+        shortest_path.append(start)
+        return shortest_path
+
+    def ucs(
+        self,
+        current: list[int],
+        final: list[list[int]],
+        grid: list[list[int]],
+        dxy: list[list[int]],
+        goal_answer: list[int],
+    ) -> list[list[float]] | list[list[int]] | None:
+        """
+        Return 2D list where optimal path is stored.
+        # >>> current = [0, 0]
+        # >>> final = [[1, 1]]
+        # >>> grid = [[0,0],[0,0]]
+        # >>> dxy = [(1, 1),(1, 0),(1, -1),(0, -1),(-1, -1),(-1, 0),(-1, 1),(0, 1)]
+        # >>> goal_answer = [1000000, 100000]
+        # >>> ucs(current,final,grid,dxy,goal_answer)
+        # [[0,2],[1,1]]
+        """
+
+        dist = [[float("inf") for _ in range(self.m)] for _ in range(self.n)]
+        visited = [[0 for _ in range(self.m)] for _ in range(self.n)]
+
+        x = current[0]
+        y = current[1]
+        dist[x][y] = 0
+        final_cnt = 0
+        heap = [[0, x, y]]
+        while len(heap) > 0:
+            heap.sort(key=lambda x: x[0])
+            d, x, y = heap[0]
+            heap.pop(0)
+            if visited[x][y]:
+                continue
+            visited[x][y] = 1
+            if [x, y] in final:
+                idxs = [ix for ix, iy in enumerate(final) if iy == [x, y]]
+                if len(idxs) > 1:
+                    print("Twice")
+                if goal_answer[idxs[0]] == 10**8:
+                    final_cnt += 1
+                    print("Increment")
+                if goal_answer[idxs[0]] > d:
+                    goal_answer[idxs[0]] = d
+                    print("Extended distance")
+                if final_cnt == len(final):
+                    path = self.get_shortest_path(current, final[0], dist, dxy)
+                    return path
+            for dx, dy in dxy:
+                rx = dx + x
+                ry = dy + y
+                if 0 <= rx < self.m and 0 <= ry < self.n and grid[rx][ry] != 1:
+                    weight = 1
+                    # diagonal_weight = 1.414
+                    # if (
+                    #     (dx == -1 and dy == -1)
+                    #     or (dx == 1 and dy == 1)
+                    #     or (dx == 1 and dy == -1)
+                    #     or (dx == -1 and dy == 1)
+                    # ):
+                    #     new_dist = d + diagonal_weight
+                    # else:
+                    new_dist = d + weight
+                    if new_dist < dist[rx][ry]:
+                        dist[rx][ry] = new_dist
+                        heap.append([new_dist, rx, ry])
+        return None
+
+    def your_algorithm(
+        self, start_point: list[int], end_point: list[int], grid: list[list[int]]
```

Comment:
Instead of `your_algorithm`, give a proper understandable name to this function.

Reply:
Given 'execute_algorithm' for running the UCS algorithm